src/awala.ts

```diff
@@ -33,3 +48,117 @@ export async function createMemberBundleRequest(
     didSucceed: true,
   };
 }
+// string will be substituted with the actual error
+export async function generateMemberBundle(
+  publicKeyId: string,
+  options: ServiceOptions,
+): Promise<Result<ArrayBuffer, string>> {
+
+  // getMemberPublicKey required memberId as a parameter
+  const memberPublicKeyModel = getModelForClass(MemberPublicKeyModelSchema, {
+    existingConnection: options.dbConnection,
+  });
+
+  // getMember orgName is a required parameter
+    const memberModel = getModelForClass(MemberModelSchema, {
+      existingConnection: options.dbConnection,
+    });
+
+    // getOrg does not return privateKeyRef and publicKeyRef
+    const orgModel = getModelForClass(OrgModelSchema, {
+      existingConnection: options.dbConnection,
+    });
+
+
+  const memberPublicKey = await memberPublicKeyModel.findById(publicKeyId);
+  if(!memberPublicKey){
+    return {
+      didSucceed: false,
+      reason: "No member public key"
+    }
+  }
+
+  const member = await memberModel.findById(memberPublicKey.memberId);
+  if(!member){
+    return {
+      didSucceed: false,
+      reason: "No member"
+    }
+  }
+  if(!member.name){
+    return {
+      didSucceed: false,
+      reason: "No member name"
+    }
+  }
+
+  const org = await orgModel.findOne({name: member.orgName});
+  if(!org){
+    return {
+      didSucceed: false,
+      reason: "No org"
+    }
+  }
+
+  const kms = await Kms.init();
+  const orgPrivateKey = await kms.retrievePrivateKeyByRef(org.privateKeyRef);
+  const orgPublicKey = await derDeserializePublicKey(org.publicKey);
+  const orgKeyPair : CryptoKeyPair = {privateKey: orgPrivateKey, publicKey: orgPublicKey};
+
+  const memberCryptoPublicKey = await derDeserializePublicKey(memberPublicKey.publicKey)
+
+  const expiryDate = addDays(new Date(), 90);
+  const orgCertificate = await selfIssueOrganisationCertificate(
+    member.orgName,
+    orgKeyPair,
+    expiryDate,
+  );
+
+  const memberCertificate = await issueMemberCertificate(
+    member.name,
+    memberCryptoPublicKey,
+    orgCertificate,
+    orgPrivateKey,
+    expiryDate,
+  );
+
+  const dnssecChain = await retrieveVeraDnssecChain(member.orgName, undefined);
```

Comment:
Two things:

- This is quite likely to throw an error, as it connects to a DNS resolver on the Internet, so it'd be good to catch any errors and do a `log.error(...)`. Otherwise, it'd be an uncaught error that would crash the process.
- I _think_ the `undefined` is unnecessary.